polaris/dataset/_base.py

```diff
@@ -116,6 +120,19 @@ def _validate_base_dataset_model(self) -> Self:
 
         return self
 
+    @model_validator(mode="after")
+    def _ensure_cache_dir_exists(self) -> Self:
+        """
+        Set the default cache dir if none and make sure it exists
+        """
+        if self._cache_dir is None:
+            dataset_id = self._md5sum if self.has_md5sum else str(uuid4())
+            self._cache_dir = str(Path(DEFAULT_CACHE_DIR) / _CACHE_SUBDIR / dataset_id)
+        fs, path = fsspec.url_to_fs(self._cache_dir)
```

Comment:
**Comment:** The `_md5sum` and `has_md5sum` properties don't exist for V2, which is why we moved this logic related to the cache dir to the subclasses. We should probably pull out a `get_cache_dir()` method instead.

```suggestion
        cache_dir = self.get_cache_dir()
        fs, path = fsspec.url_to_fs(cache_dir)
```

Reply:
Let's just not use the md5 checksum for the cache dir. Fewer chances of problems that was as well.